### PR TITLE
Deliverable 1

### PR DESCRIPTION
**Pull request by angelazb**

### CODE REVIEW

**Review by fifi-hobby — APPROVED** [review]

File: deliverable/1/backend.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+These backends support rendering for user interfaces of applications. 
+
+- `BackendGTK3`, which contains `FigureManagerGTK3` and `FigureCanvasGTK3` (implementations of `FigureManagerBase`, and `FigureCanvasBase`, respectively). 
+- `BackendMac`, which contains `FigureManagerMac` and `FigureCanvasMac` (implementations of `FigureManagerMac`, and `FigureCanvasBase`, respectively). 
+
+### Hard Copy Backends
+
+These backends support rendering for standalone files such as PDF. Hard copy backends are further categorized to Raster, and Vector.
+
```
Comment (JacobChamberlain) [inline deliverable/1/backend.md:42]: The Sections below (Raster, Vector) do not seem like they need a big title. They also seem to be repetitive, maybe we should do something like:
### Implementations of FigureCanvasBase
- /Raster/BackendCario/FigureCanvasCario
- /Raster/BackendAGG/FigureCanvasAGG
- /Vector/BackendSVG/FigureCanvasSVG
- /Vector/BackendPdf/FigureCanvasPdf

File: deliverable/1/deliverable_1.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+The process we chose is an heavily modified "waterfall" process incorporating elements from test-driven development (especially useful for bug fixing and large unfamiliar projects) and from Agile (for work estimation and some tracking). We also incorporated elements from Kanban to minimize clutter on our Trello board and to ensure that tasks are not left abandoned for unreasonable periods, especially those upon which others are dependent.
+
+To see a full explanation of the process we designed (and the reasons why) see [here](process.md).
+
+## Architecture Analysis
+
+Documentation [source available here](https://github.com/matplotlib/matplotlib/tree/master/doc) and [hosted here](https://matplotlib.org/3.1.1/contents.html).
+
+Primarily, matplotlib is a library, whose `import`able elements are under the [matplotlib](https://github.com/matplotlib/matplotlib/tree/master/lib/matplotlib) module.
```
Comment (JacobChamberlain) [inline deliverable/1/deliverable_1.md]: `import`able seems like a mistake

File: deliverable/1/deliverable_1.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+### The 3-tiered Architecture
+
+The current architecture of matplotlib is based around creating, rendering, and updating `Figure` objects. It consists of three stacked up layers. The **encapsulation** of Matplotlib is in such a way that each layer in the stack is only aware of and interacts with lower levels which results in an appropriate distribution of complexity. Each layer is responsible for a particular functionality such as event-based interactions, visual component abstraction and creation, or stateful user interaction ![Top level Diagram](./img/UML_Top_Level.svg)
```
Comment (JacobChamberlain) [inline deliverable/1/deliverable_1.md]: The `!` mark before Top Level Diagram

File: deliverable/1/peer_review.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+- [ ] Passes CI builds at least as well as before change
+- [ ] Task description satisfied accurately and in entirety
+- [ ] Sufficient tests for new or previously broken functionality
+- [ ] Formatted according to PEP8
+- [ ] No obvious bugs or oversight
+- [ ] Meaningful variable names
+- [ ] Any new dependencies documented and added to Dockerfile
+- [ ] No new pylint errors or warnings (unless justified and approved)
+- [ ] External Documentation (docstrings) for all new functions, classes. modules and methods
```
Comment (JacobChamberlain) [inline deliverable/1/peer_review.md]: `classes.` -> `classes,`

Comment (JacobChamberlain) [reply deliverable/1/peer_review.md]: also `modules and methods` -> `modules, and methods`

File: deliverable/1/process.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+## Continuous Integration
+
+All branches will be built and run with the Travis and Circle Continuous Integration tools. For both tools, configuration files and instructions for activating the GitHub add-ons are provided by the Matplotlib development team.
+
+Although they (the matplotlib developers) do not require or even strongly recommend that contributors set up CI on their own forks, we would like to have the best chance possible of having our PR merged into the master matplotlib repo. For that to be the case, we would like to know if any changes we make cause the build to fail, immediately. Their current recommended strategy involves kicking back PRs that break the build, but due to our time constraint we find this unacceptable.
+
+## Branching
+
+Our branching strategy is simple enough: where possible, work on a unique branch for your task, named after the task.
```
Comment (JacobChamberlain) [inline deliverable/1/process.md:62]: Somewhere in this section (not sure about under Branching) we should talk about re basing or other methods to simplify the commit history.

**Review by JacobChamberlain — CHANGES_REQUESTED** [review]
I left some comments in the files, please take a look and justify keeping it the way it is or make the changes before merging.

File: deliverable/1/backend.md (hunk 1 above)
Comment (fifi-hobby) [reply deliverable/1/backend.md:42]: Changed the headers of Raster and Vector and made them smaller.

File: deliverable/1/deliverable_1.md (hunk 2 above)
Comment (fifi-hobby) [reply deliverable/1/deliverable_1.md]: Changed `import`able to importable.

File: deliverable/1/deliverable_1.md (hunk 3 above)
Comment (fifi-hobby) [reply deliverable/1/deliverable_1.md]: This is just syntax for the image, but I have included a break between the paragraph and the image.

File: deliverable/1/process.md (hunk 5 above)
Comment (fifi-hobby) [reply deliverable/1/process.md:62]: Added a description for rebasing strategy under "Merging to master" section.

File: deliverable/1/peer_review.md (hunk 4 above)
Comment (fifi-hobby) [reply deliverable/1/peer_review.md]: Fixed grammar issue.

**Review by JacobChamberlain — APPROVED** [review]

**Pull request merged**